app/src/main/java/com/asfoundation/wallet/util/TransferParser.java

```diff
@@ -128,10 +128,11 @@ private String getPayload(ERC681 payment) throws UnsupportedEncodingException {
   }
 
   private TransactionData retrieveData(ERC681 payment) throws UnsupportedEncodingException {
-    String data = new String(Hex.decode(payment.getFunctionParams()
+    String data = payment.getFunctionParams()
```

Comment:
This change leads to a null pointer exception anyway. Should consider the URI null data as a an empty object?